rllib/offline/dataset_writer.py

```diff
@@ -50,7 +51,15 @@ def __init__(
         ), "output_config.path must be specified when using Dataset output."
 
         self.format = output_config["format"]
-        self.path = os.path.abspath(os.path.expanduser(output_config["path"]))
+        parsed_path = urllib.parse.urlparse(output_config["path"])
+        self.is_filepath = parsed_path.scheme == ""
```

Comment:
@simonsays1980 Afaics, urlparse sets `scheme=""` even in cases of web adresses.
-> https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse
This means that things like the following can happen:

```
> parsed_path = urllib.parse.urlparse("www.google.com")
# parsed_path.scheme == ""  is now True
> pathlib.Path(parsed_path.path).expanduser().resolve()
PosixPath('/Users/artur/code/ray/www.google.com')
```

How do we best deal with this?

Reply:
Imo, this should be okay, as the `scheme` should refer to a protocol like `http`, `s3`, `gs`, etc. `www` is the set of linked hypertext documents that can be viewed in a browser. If a user wants to write to cloud storage he/she must provide the protocol that should be used. Using a web address like `www.google.com` therefore should fail in the `DataWriter` - we want this imo.

We can think of providing the user in such a case with guiding messages like: _"The link you provided started with `www`, however `DataWriter` needs a protocol like `https`, `s3`, etc. in front of any cloud storage address. An example for S3 cloud storage would look like `s3://`"_

Reply:
@ArturNiederfahrenhorst I added now the `try_import_arrow()`. It might be that we need to also check for a certain version of `pyarrow`, afaicr I needed to downgrade `pyarrow` the last time to make it run.

Reply:
@ArturNiederfahrenhorst How do we want to proceed here?